### ISSUE TITLE
amphtml-validator: severity reported as WARNING instead of ERROR for invalid JSON in script tag.

### ISSUE THREAD

**Issue by ashu82492** [ampproject/amphtml#26655]
We use amphtml-validator https://www.npmjs.com/package/amphtml-validator version 1.0.29

I have an **error** on my AMP Page (in Browser Console) that stats `[AmpAnalytics <unknown id>] Failed to parse <script> contents. Is it valid JSON?`

But with amphtml-validator, I am only getting this with severity **WARNING** 
`The script tag contains invalid JSON that cannot be parsed`.

In this case, `error.severity` (in the code below) should have been ERROR.

Code that I am using to validate the AMP URL

```
request(url, function (error, response, body) {
    let result = validator.validateString(body);
    let resp = ""
    if(response){
        resp = response.statusCode
    }
    ((result.status === 'PASS') ? console.log : console.error)("\nTesting: ", url, "\nResponse Code: ", resp,  "\n"+ result.status);
    for (var ii = 0; ii < result.errors.length; ii++) {
        var error = result.errors[ii];
        var msg = 'line ' + error.line + ', col ' + error.col + ': ' + error.message;
        if (error.specUrl !== null) {
            msg += ' (see ' + error.specUrl + ')';
        }
        ((error.severity === 'ERROR') ? console.error : console.warn)(msg);
    }
    if (result.status !== 'PASS') failPipeline = true;
    callback(null);
}

```
The issue is that It should be an **error** instead of a **warning**. `result.status` status is `PASS`, it should be `FAIL`.

## How do we reproduce the issue?

Use this tag (the JSON is missing a `}`) in html
```
<amp-analytics type="googleanalytics">
    <script type="application/json">
        {
            "vars": {
                "account": "XXXXXXX"
            },
            "extraUrlParams": {
                "pageSourceType": "AMP"
            },
            "triggers": {
                "trackPageview": {
                    "on": "visible",
                    "request": "pageview",
                    "vars": {
                        "title": "@Html.Raw(Model.PageMetaTags.Title)"
                    }
                }
            }
        
    </script>
</amp-analytics>
```

**Comment by honeybadgerdontcare** [comment]
This is intentionally a `warning` instead of an `error` because it would make a large number of documents that were valid to become invalid. That type of change is considered a breaking change and would need to go through an [I2D and I2R](https://github.com/ampproject/amphtml/issues/new/choose) process.

**Comment by cw-ashutoshsharma** [comment]
@honeybadgerdontcare I did not get your point, `it would make a large number of documents that were valid to become invalid`.

The script tag has invalid Json and amphtml-validator should report only invalid ones as an error.

**Comment by honeybadgerdontcare** [comment]
This was a relatively new feature of the validator that reported that JSON was not valid. That's why we made it a warning. If it was done at the release of AMP then it would have been an error.

If we had made it an error now then lots of AMP documents would suddenly become invalid and AMP Caches such as Google, Bing and others would not serve those documents. It would be bad for developers and the ecosystem. So it's a warning and still discoverable in AMP devtools.

There is the I2D and I2R process to assess if this is something the AMP community wants to do, but I don't think it's time yet due to the number of previously valid AMP documents would become invalid.
